code/controllers/subsystem/SSjobs.dm

```diff
@@ -529,7 +529,7 @@ SUBSYSTEM_DEF(jobs)
 				if(istype(G) && !G.prescription)
 					G.upgrade_prescription()
 					H.update_nearsighted_effects()
-	if(late_arrivals_spawning)
+	if(late_arrivals_spawning && !joined_late)
 		H.forceMove(pick(GLOB.latejoin))
 	if(!joined_late && drunken_spawning)
```

Comment:
Can you make this an early return? Can you also make it return if the job is an admin job (this can be checked with job.admin_only)?
```
if(joined_late)
    return H
```

Reply:
done